tensorcircuit/applications/optimization.py

```diff
@@ -0,0 +1,328 @@
+        params = backend.implicit_randn(shape=[2 * nlayers], stddev=0.5)
+        if vvag == True:
+            loss_val_grad = tc.backend.vvag(
+                loss_val_grad, argnums=0, vectorized_argnums=0
+            )
+            params = backend.implicit_randn(shape=[ncircuits, 2 * nlayers], stddev=0.1)
+            # If init_params is not provided, initialize the parameters randomly.
+            # If vvag flag is set to True, use vectorized variational adjoint gradient (vvag) with multiple circuits.
+    else:
+        params = init_params
+        # If init_params is provided, use the provided parameters.
+    # Initialize the parameters for the ansatz circuit.
+
+    if jit == True:
+        loss_val_grad = backend.jit(loss_val_grad, static_argnums=(1, 2))
+        # Use just-in-time compilation (jit) if jit flag is set to True.
+    # This can improve the performance by pre-compiling the loss and gradients function.
+
+    opt = backend.optimizer(tf.keras.optimizers.Adam(learning_rate))
+    # Define the optimizer (Adam optimizer) with the specified learning rate.
+
+    for i in range(iterations):
+        loss, grads = loss_val_grad(params)
+        # Calculate the loss and gradients using the loss_val_grad_jit function.
+
+        params = opt.update(grads, params)
+        # Update the parameters using the optimizer and gradients.
+
+        if callback is not None:
+            callback(loss, params)
+        # Execute the callback function with the current loss and parameters.
+
+    return params
+    # Return the optimized parameters for the ansatz circuit.
+
+
+def cvar_value(r: list, p: list, percent: float) -> float:
+    """
+    Calculate the Conditional Value at Risk (CVaR) according to the measurement results.
+
+    :param r: The results showing after measurements.
+    :param p: Probabilities corresponding to each result.
+    :param percent: The cut-off percentage of CVaR.
+    :return: The calculated CVaR value.
+    """
+    sorted_indices = np.argsort(r)
+    p = np.array(p)[sorted_indices]
+    r = np.array(r)[sorted_indices]
+
+    sump = 0.0  # The sum of probabilities.
+    count = 0
+    cvar_result = 0.0
+
+    # Iterate over the sorted results and calculate CVaR.
+    while sump < percent:
+        if round(sump + p[count], 6) >= percent:
+            # Add the remaining portion of the last result that exceeds the cut-off percentage.
+            cvar_result += r[count] * (percent - sump)
+            count += 1
+            break
+        else:
+            # Add the entire result to the CVaR calculation.
+            sump += p[count]
+            cvar_result += r[count] * p[count]
+            count += 1
+
+    cvar_result /= percent
+    return cvar_result
+
+def cvar_from_circuit(
+    circuit: Circuit, nsamples: int, Q: Tensor, alpha: float
+) -> float:
+    """
+    Directly calculate the Conditional Value at Risk (CVaR) from a circuit.
+    The CVaR depends on a bunch of measurements.
```

Comment:
'depends on a bunch of measurements' sounds too informal. Can we express this more formally?